pkg/helper/podHelper.go

```diff
@@ -0,0 +1,297 @@
+		Command: command,
+		Stdin:   false,
+		Stdout:  true,
+		Stderr:  true,
+		TTY:     true,
+	}
+	req.VersionedParams(
+		option,
+		scheme.ParameterCodec,
+	)
+	exec, err := remotecommand.NewSPDYExecutor(restConfig, "POST", req.URL())
+	if err != nil {
+		return "", "", errors.Errorf("Failed executing command on throwaway pod %s", podName)
+	}
+
+	buf := &bytes.Buffer{}
+	errBuf := &bytes.Buffer{}
+
+	err = exec.StreamWithContext(context.Background(), remotecommand.StreamOptions{
+		Stdout: buf,
+		Stderr: errBuf,
+	})
+	if err != nil {
+		return "", "", errors.Errorf("Failed executing command on throwaway pod %s", podName)
+	}
+
+	return buf.String(), errBuf.String(), nil
+}
+
+// ExecuteRemoteContainerCommand exec command on specific pod and wait the command's output.
+func (p PodExecutor) ExecuteRemoteContainerCommand(ns string, podName string, container string, command []string) (string, string, error) {
+
+	kubeClient, restConfig, err := getClient()
+	if err != nil {
+		return "", "", errors.Errorf("Failed to get client during throwaway pod execuction command: %s", err)
+	}
+
+	req := kubeClient.CoreV1().RESTClient().Post().Resource("pods").Name(podName).
+		Namespace(ns).SubResource("exec")
+	option := &v1.PodExecOptions{
+		Command:   command,
+		Stdin:     false,
+		Stdout:    true,
+		Stderr:    true,
+		TTY:       true,
+		Container: container,
+	}
+	req.VersionedParams(
+		option,
+		scheme.ParameterCodec,
+	)
+	exec, err := remotecommand.NewSPDYExecutor(restConfig, "POST", req.URL())
+	if err != nil {
+		return "", "", errors.Errorf("Failed executing command on throwaway pod %s", podName)
+	}
+
+	buf := &bytes.Buffer{}
+	errBuf := &bytes.Buffer{}
+
+	err = exec.StreamWithContext(context.Background(), remotecommand.StreamOptions{
+		Stdout: buf,
+		Stderr: errBuf,
+	})
+	if err != nil {
+		return "", "", errors.Errorf("Failed executing command on throwaway pod %s", podName)
+	}
+
+	return buf.String(), errBuf.String(), nil
+}
+
+func getClient() (*kube.Clientset, *restclient.Config, error) {
+
+	kubeCfg := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		clientcmd.NewDefaultClientConfigLoadingRules(),
+		&clientcmd.ConfigOverrides{},
+	)
+	restCfg, err := kubeCfg.ClientConfig()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	kubeClient, err := kube.NewForConfig(restCfg)
+	if err != nil {
+		return nil, nil, errors.Errorf("Failed to generate new client for throwaway pod %s", err)
+	}
+	return kubeClient, restCfg, nil
+}
+
+func CreateRedisThrowAwayPod(k8sclient client.Client, namespace string) (*v1.Pod, error) {
+	// Create throwaway redis deployment
+	systemRedisPod := throwAwayRedis(namespace)
+
+	err := k8sclient.Create(context.TODO(), systemRedisPod)
+	if err != nil {
+		if !k8serrors.IsAlreadyExists(err) {
+			// if it's other error than "AlreadyExists"
+			return systemRedisPod, err
+		}
+	}
+
+	// Wait for deployment to become ready
+	err = wait.Poll(time.Second*5, time.Minute*3, func() (done bool, err error) {
```

Comment:
See above comment about replacing deprecated `wait.Poll` with `wait.PollUntilContextTimeout`

Reply:
If it's not a big deal, I'd rather stick to the old version for now and refactor when required.